src/TokenParser/FromTokenParser.php

```diff
@@ -47,7 +47,7 @@ public function parse(Token $token)
         $stream->expect(/* Token::BLOCK_END_TYPE */ 3);
 
         $var = new AssignNameExpression($this->parser->getVarName(), $token->getLine());
-        $node = new ImportNode($macro, $var, $token->getLine(), $this->getTag());
+        $node = new ImportNode($macro, $var, $token->getLine(), $this->getTag(), $this->parser->isMainScope());
```

Comment:
how does `parser->isMainScope()`  behave when parsing an `embed` ?

Reply:
When using an `embed` tag, the parser is reset to an initial stage, so, it behaves like it would for a new template, which is what we expect (see tests).